### PR TITLE
Add destinations to saas dependency callback

### PR DESCRIPTION
**Pull request by alexpenev-s**
**Description**

Changes proposed in this pull request:
Add destinations to saas dependency callback

**Related issue(s)**

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/Makefile [hunk 1]
```diff
@@ -43,6 +43,8 @@ deploy-on-k3d: build-for-k3d
 	kubectl patch -n $(NAMESPACE) cronjob/compass-ord-aggregator -p '{"spec":{"jobTemplate":{"spec":{"template":{"spec":{"containers":[{"name":"aggregator","imagePullPolicy":"Always"}]}}}}}}'
 	kubectl set image -n $(NAMESPACE) deployment/compass-tenant-fetcher tenant-fetcher=k3d-kyma-registry:5001/compass-director:latest
 	kubectl rollout restart -n $(NAMESPACE) deployment/compass-tenant-fetcher
+	kubectl set image -n $(NAMESPACE) deployment/compass-tenant-fetcher tenant-fetcher=k3d-kyma-registry:5001/compass-director:latest
```
Comment (StefanCholakov) [inline components/director/Makefile:46]: Why this is added for a second time?

File: chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{{- if eq .Values.global.externalServicesMock.tenantDependenciesConfig.manage true }}
```
Comment (StefanCholakov) [inline chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml:1]: Can you please wrap the condition in if in brackets, like this:
{{- if (eq .Values.global.externalServicesMock.tenantDependenciesConfig.manage true) }}

File: chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  name: {{ .Values.global.tenantFetcher.dependenciesConfig.secretName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+    release: {{ .Release.Name }}
+stringData:
+  keyConfig: |
+    {
+    "eu-1": "{\n \"credentials\": {\n \"xsappname\": \"xsappname1\"\n}\n}",
```
Comment (StefanCholakov) [inline chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml:14]: How these xsappname values can be changed in production? Currently, there are two hardcoded xsappnames for eu-1 and eu-2.

File: components/director/Makefile (hunk 1 above)
Comment (alexpenev-s) [reply components/director/Makefile:46]: It's a mistake. Probably a bad rebase. Fixed.

File: chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml (hunk 3 above)
Comment (alexpenev-s) [reply chart/compass/charts/external-services-mock/templates/tenant-dependencies-configmap.yaml:14]: These are generated in production as follows:
1. Create destination instances for each landscape (in the lss scripts) and store the credentials in vault
2. The deploy Jenkins job takes the credentials from vault and creates a secret in the format `{ "region": { "credentials": .... } }`.
3. The destination fetcher mounts the secret and parses the xsappnames. 
4. Destinations callback returns the proper xsappname given the region (passed as a url parameter)

**Review by desislavaa — APPROVED** [review]

**Review by nkanchev — DISMISSED** [review]

**Review by desislavaa — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]

**Pull request merged**